pkg/agent/cniserver/pod_configuration_test.go

```diff
@@ -0,0 +1,461 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cniserver
+
+import (
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"antrea.io/libOpenflow/openflow15"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	openflowtest "antrea.io/antrea/pkg/agent/openflow/testing"
+	"antrea.io/antrea/pkg/agent/types"
+	"antrea.io/antrea/pkg/util/channel"
+)
+
+var (
+	podIfName           = "test"
+	podIPs              = []net.IP{net.ParseIP("192.168.9.10")}
+	podMac, _           = net.ParseMAC("00:15:5D:B2:6F:38")
+	podInfraContainerID = "261a1970-5b6c-11ed-8caf-000c294e5d03"
+
+	pod = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testPodNameA,
+			Namespace: testPodNamespace,
+		},
+		Spec: corev1.PodSpec{
+			NodeName: nodeName,
+		},
+	}
+
+	portStatusMsg = &openflow15.PortStatus{
+		Reason: openflow15.PR_MODIFY,
+		Desc: openflow15.Port{
+			PortNo: 1,
+			Length: 72,
+			Name:   []byte(fmt.Sprintf("%s\x00", podIfName)),
+			State:  openflow15.PS_LIVE,
+		},
+	}
+
+	podIfaceConfig = interfacestore.InterfaceConfig{
+		InterfaceName: podIfName,
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			PodNamespace: testPodNamespace,
+			PodName:      testPodNameA,
+			ContainerID:  podInfraContainerID,
+		},
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "test-port-uuid",
+		},
+		IPs: podIPs,
+		MAC: podMac,
+	}
+)
+
+type mockClients struct {
+	kubeClient       *fakeclientset.Clientset
+	informerFactory  informers.SharedInformerFactory
+	localPodInformer cache.SharedIndexInformer
+	podLister        corelisters.PodLister
+	ofClient         *openflowtest.MockClient
+	recorder         *record.FakeRecorder
+}
+
+func newMockClients(ctrl *gomock.Controller, nodeName string, objects ...runtime.Object) *mockClients {
+	kubeClient := fakeclientset.NewClientset()
+	for _, obj := range objects {
+		if obj != nil {
+			kubeClient.Tracker().Add(obj)
+		}
+	}
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
```

Comment:
you are not using the informer factory anywhere as far as I can tell, it should be removed

(this is because you use a filtered Pod informer, which presumably cannot be created from an informer factory)